code/modules/mob/living/simple_animal/friendly/trader.dm

```diff
@@ -0,0 +1,242 @@
+/**
+ * # Trader
+ *
+ * A mob that has some dialogue options with radials, allows for selling items and buying em'
+ *
+ */
+
+/mob/living/simple_animal/hostile/retaliate/trader
+	name = "Trader"
+	desc = "Come buy some!"
+	icon = 'icons/mob/simple_human.dmi'
+	icon_state = "faceless"
+	maxHealth = 200
+	health = 200
+	melee_damage_lower = 10
+	melee_damage_upper = 10
+	attack_verb_continuous = "punches"
+	attack_verb_simple = "punch"
+	attack_sound = 'sound/weapons/punch1.ogg'
+	del_on_death = TRUE
+	loot = list(/obj/effect/mob_spawn/human/corpse)
+	atmos_requirements = list("min_oxy" = 5, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 1, "min_co2" = 0, "max_co2" = 5, "min_n2" = 0, "max_n2" = 0)
+	unsuitable_atmos_damage = 2.5
+	casingtype = /obj/item/ammo_casing/shotgun/buckshot
+	wander = FALSE
+	ranged = TRUE
+	combat_mode = TRUE
+	move_resist = MOVE_FORCE_STRONG
+	mob_biotypes = MOB_ORGANIC|MOB_HUMANOID
+	sentience_type = SENTIENCE_HUMANOID
+	speed = 0
+	stat_attack = HARD_CRIT
+	robust_searching = TRUE
+	check_friendly_fire = TRUE
+	///Sound used when item sold/bought
+	var/sell_sound = 'sound/effects/cashregister.ogg'
+	///Phrases used when you talk to the NPC
+	var/list/lore = list("Hello! I am the test trader.",
+						"Oooooooo~!"
+						)
+	///Associated list of items the NPC sells with how much they cost.
+	var/list/products = list(/obj/item/food/burger/ghost = 200)
+	///Associated list of items able to be sold to the NPC with the money given for them.
+	var/list/wanted_items = list(/obj/item/ectoplasm = 100)
+	///Phrase said when NPC finds none of your inhand items in wanted_items.
+	var/itemrejectphrase = "Sorry, I'm not a fan of anything you're showing me. Give me something better and we'll talk."
+	///Phrase said when you cancel selling a thing to the NPC.
+	var/itemsellcancelphrase = "What a shame, tell me if you changed your mind."
+	///Phrase said when you accept selling a thing to the NPC.
+	var/itemsellacceptphrase = "Pleasure doing business with you."
+	///Phrase said when the NPC finds an item in the wanted_items list in your hands.
+	var/interestedphrase = "Hey, you've got an item that interests me, I'd like to buy it, I'll give you some cash for it, deal?"
+	///Phrase said when the NPC sells you an item.
+	var/buyphrase = "Pleasure doing business with you."
+	///Phrase said when you have too little money to buy an item.
+	var/nocashphrase = "Sorry adventurer, I can't give credit! Come back when you're a little mmmmm... richer!"
+
+/mob/living/simple_animal/hostile/retaliate/trader/attack_hand(mob/user)
+	. = FALSE
+	if(SEND_SIGNAL(src, COMSIG_ATOM_ATTACK_HAND, user) & COMPONENT_CANCEL_ATTACK_CHAIN)
```

Comment:
Why does this not call `..()`? Signals should realistically only fire from one place.

Reply:
it dont work

Reply:
CC @ninjanomnom if this is okay, or if there's a better solution.

Reply:
There si a better solution this is an awful work around. What didn't work about calling. =..()

Reply:
You'll need to explain more in particular what you want to prevent but the interact() proc is likely what you want